entities/StandardGameState.cs

```diff
@@ -0,0 +1,313 @@
+using System;
+using System.Collections.Generic;
+
+
+namespace LightChess
+{
+    /*
+    This class represents the game state. Game state can be uniquely identified by FEN notation.
+    In addition, the + operator is overloaded to get the next game state after move.
+
+    Please refer to FEN notation (https://en.wikipedia.org/wiki/Forsyth%E2%80%93Edwards_Notation). 
+
+    TODO: pawn promotion is not implemented
+    */
+
+    public class StandardGameState: GameState
+    {
+    
+        public StandardBoard Board;
+        public Color ActiveColor;
+        public Color EnemyColor;
+        public List<Castle> AvaialbleCastles;
+        public Cell? EnPassantCell;
+        public int HalfmoveNumber;
+        public int FullmoveNumber;
+
+        public StandardGameState(
+            StandardBoard board,
+            Color color,
+            List<Castle> avaialbleCastles,
+            Cell? enPassantCell,
+            int halfmoveNumber,
+            int fullmoveNumber
+        )
+        {
+        Board = board;
+        ActiveColor = color;
+        EnemyColor = color.Change();
+        AvaialbleCastles = avaialbleCastles;
+        EnPassantCell = enPassantCell;
+        HalfmoveNumber = halfmoveNumber;
+        FullmoveNumber = fullmoveNumber;
+        }
+    
+        /// <summary>
+        /// Get next game state.
+        /// </summary>
+        /// <param name="gameState">
+        /// The start/initial game state.
+        /// </param>
+        /// <param name="move">
+        /// The move to perform. The move must be valid/possible.
+        /// </param>        
+        /// <returns>
+        /// The next game state. 
+        /// </returns>
+        public static StandardGameState operator+(StandardGameState gameState, 
+                                                  Move move)
+        {
+            
+            var piece = (Piece)gameState.Board[move.StartCell];
+            var deltaX = move.EndCell.X - move.StartCell.X;
+            StandardBoard nextBoard;
+
+            // Castle.
+            if (piece.Type == PieceType.King || deltaX >= 2)
+            {
+                nextBoard = PerformCastle(gameState.Board, move);
+            }
+            // Enpassant move.
+            else if (gameState.EnPassantCell != null 
+                    &&  move.EndCell == (Cell)gameState.EnPassantCell)
+            {
+                nextBoard = PerformEnPassantMove(gameState.Board, move);
+            }
+            // Simple move.
+            else
```

Comment:
Here also I would prefer writing `} else {`, but again, we can agree on a style guide.